skill-tree/src/main/java/com/RDS/skilltree/utils/JWTUtils.java

```diff
@@ -0,0 +1,88 @@
+package com.RDS.skilltree.utils;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.Jwts;
+import jakarta.annotation.PostConstruct;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.AuthenticationCredentialsNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.Date;
+import java.util.function.Function;
+
+@Component
+@Slf4j
+public class JWTUtils {
+
+    @Value("${jwt.rds.public.key}")
+    private String publicRDSKeyString;
+    private KeyFactory keyFactory;
+
+    @PostConstruct
+    public void init() throws NoSuchAlgorithmException {
+        keyFactory = KeyFactory.getInstance("RSA");
+    }
+
+    /**
+     * Converts the given public key string to an RSAPublicKey object.
+     *
+     * @param publicKeyString the public key string to be converted
+     * @return the RSAPublicKey object converted from the public key string
+     */
+    private RSAPublicKey convertToRSAPublicKey(String publicKeyString) {
+        try {
+            publicKeyString = publicKeyString.replace("-----BEGIN PUBLIC KEY-----", "")
+                    .replace("-----END PUBLIC KEY-----", "")
+                    .replaceAll("\\s", "");
+
+            byte[] publicKeyBytes = Base64.getDecoder().decode(publicKeyString);
+            X509EncodedKeySpec keySpec = new X509EncodedKeySpec(publicKeyBytes);
+            return (RSAPublicKey) keyFactory.generatePublic(keySpec);
+        } catch (InvalidKeySpecException exception) {
+            throw new RuntimeException("Invalid RSA key");
+        }
+    }
+
+    private Claims extractAllClaims(String token) {
+        return Jwts
+                .parser()
+                .setSigningKey(convertToRSAPublicKey(publicRDSKeyString))
```

Comment:
The value of `convertToRSAPublicKey(publicRDSKeyString)` is always going to be the same as the `publicRDSKeyString` is going to be unchanged.
Rather than using the method call here to convert the key every time `extractAllClaims` is called (which will be for every request that has a Token in header), can we optimise this?

Reply:
was also thinking of it, but didn't get how I can do a calculation and store the result of it here 
 can you please suggest me, something I need to look for to achieve this

Reply:
Class initialization/construction!